### ISSUE TITLE
Usage react-url-query with react-router v4 throwing error

### ISSUE THREAD

**Issue by lineldcosta** [pbeshai/react-url-query#37]
I am using react-url-query with react-router v4 as shown in the example but its throwing **No history provided to react-url-query. Please provide one via configureUrlQuery.** error

**Comment by pbeshai** [comment]
Are you sure you're using the RouterToUrlQuery component in your code?

e.g.

```js
import { RouterToUrlQuery } from 'react-url-query';
import Router from 'react-router/BrowserRouter';

ReactDOM.render(
  <Router>
    <RouterToUrlQuery>
      <App />
    </RouterToUrlQuery>
  </Router>,
  document.getElementById('root')
);
```

(see [index.js](https://github.com/pbeshai/react-url-query/blob/master/examples/react-router-v4-and-redux/src/index.js) in the example)

**Comment by lineldcosta** [comment]
Yes same thing i copied from your examples.. If you get time once try with new react-router-dom. I tried many times but i got the same results, then with no other options i had to copy the logic from your src folder without using this library.

**Comment by pbeshai** [comment]
Thanks, I will take a look to see if I can reproduce it.

**Comment by joka828** [comment]
Same problem here, still no solution?

**Comment by leonelgalan** [comment]
@lineldcosta and @joka828? Do you still have access to an App that exhibits this issue? I have a slight variation of it where I use `ConnectedRouter` instead (from _connected-react-router_) and I noticed that the context passed to `RouterToUrlQuery` is and _Empty Object_ and before getting the error you guys posted, I get a warning saying: "RouterToUrlQuery: `router` object not found in context. Not configuring history for react-url-query."

- Do you see the same warning before the error? On Development.
- Could you explore what's the context `RouterToUrlQuery` is getting?

I'm trying to confirm it's not `ConnectedRouter` causing the issue before I post my own. 

\* On Production the warning isn't shown and the entire app crashes in the last because router is undefined. In _lib/react/RouterToUrlQuery.js_:

```js
      if (process.env.NODE_ENV === 'development' && !router) {
        // eslint-disable-next-line
        console.warn('RouterToUrlQuery: `router` object not found in context. Not configuring history for react-url-query.');
        return;
      }

      var history = void 0;

      if (router.history && router.history.push && router.history.replace) {
        history = router.history;
      } # ...
```

**Comment by jjPlusPlus** [comment]
I am also having this problem... @leonelgalan I see the same warning, and after inspection I see that the context is empty `router: undefined`. Following the example to the letter, I've noticed something that throws me off about `import Router from 'react-router/BrowserRouter';`. My import looks like `import { BrowserRouter as Router } from "react-router-dom";`. If I tried to import the router directly from react-router/BrowerRouter it would throw an import error... So i wonder what is different between my setup and the example. You also mentioned that you are using _connectedRouter_- Well, I'm using a Firebase HOC so maybe for both of us the external library is causing the problem...
```
ReactDOM.render(
  <Provider store={store}>
    <ReactReduxFirebaseProvider {...reactReduxFirebaseProps}>
      <Router>
        <RouterToUrlQuery>
          <App />
        </RouterToUrlQuery>
      </Router>
    </ReactReduxFirebaseProvider>
  </Provider>
  , document.getElementById('root')
);

```

**Comment by jjPlusPlus** [comment]
Ah, yes, okay, switching back to react-router-dom & react-router 4.1.1 fixes it for my implementation. I will look into this further to see why context is not set on 5.1.1.

**Comment by fatpandaria** [comment]
I got the same problem,while set to react-router to 4.1.1 didn't work for me.

**Comment by leonelgalan** [comment]
@fatpandaria, would you share with us some code? Is router being passed in the context? What other versions are you using?

**Comment by fatpandaria** [comment]
@leonelgalan it's like this,I am using `create-react-app` for the app,while using `react-router-dom@5.0.1` for routes and `react-loadable` for code splitting.I configure the routes in a separate file.I bypassed this issue by passing the `history` to `configureUrlQuery` in my component.I am using the `react-loadable`, so I am not sure whether I put the `RouterToUrlQuery` in the right place;Here is the code I used:

The src/index.js is like this:
```js
// other codes...
import {Routes} from './routes/index';
ReactDOM.render(
    <Provider store={store}>
        {Routes}
    </Provider>
    , document.getElementById('root'));
```

The routes/index.js is like this:

``` js
import React from 'react';
import Loadable from 'react-loadable';
import MyLoadingComponent from '../components/common/MyLoadingComponent';
import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
import { RouterToUrlQuery } from 'react-url-query';

let AsyncHome = Loadable({
    loader: () => import('../containers/Home'),
    loading: MyLoadingComponent,
})
let AsyncChild = Loadable({
        loader: () => import('../containers/Child'),
        loading: MyLoadingComponent,
    })
const routes = [
    {
        path: '/',
        component: AsyncHome,
        children: [
            {
                path: '/child',
                component: AsyncChild,
            },
        ]
    },

]
function AlignRoute(routeProps) {
    return (
        <Route
            path={routeProps.path}
            render={
                props => (
                    <routeProps.component {...props} routes={Array.isArray(routeProps.children) && routeProps.children.map((childRoute, index) => (
                        <AlignRoute key={index} {...childRoute} />
                    ))} />
                )
            }
        />
    )
}

const Routes = (<Router>
        <Switch>
            <RouterToUrlQuery>
                {routes.map((route, index) => (
                    <AlignRoute key={index} {...route} />
                ))}
            </RouterToUrlQuery>
        </Switch>
    </Router>)

        
export {
    Routes
};
```

**Comment by leonelgalan** [comment]
@fatpandaria, for are using _react-router-dom@5.0.1_, but as you can see for previous comments, you have to stay on 4.1.1 for now. You said previously that it didn't work, but you didn't say how it didn't work. Once you go back to 4.1.1, is `router` defined in the context inside `RouterToUrlQuery`?

**Comment by lemmingapex** [comment]
react-url-query does not functioning when using react-router-dom version 5.0.1.  **No history provided to react-url-query. Please provide one via configureUrlQuery.** is shown in the console and there does not appear to be a valid history object as react-url-query expects.  react-router-dom version 5.0.1 was released Jun. 4.  When can we expect this to be fixed?

**Comment by UberMouse** [comment]
I've created PR #64 which updates the library to work correctly with React Router v5.
